source/distributed/cell_weights.cc

```diff
@@ -171,7 +267,113 @@ namespace parallel
+    return weighting_function(cell, dof_handler.get_fe(fe_index));
+  }
+
+
+
+  // ---------- deprecated functions ----------
+
+  template <int dim, int spacedim>
+  CellWeights<dim, spacedim>::CellWeights(
+    const hp::DoFHandler<dim, spacedim> &dof_handler)
+    : dof_handler(&dof_handler)
+  {
+    triangulation =
+      dynamic_cast<const parallel::TriangulationBase<dim, spacedim> *>(
+        &(this->dof_handler->get_triangulation()));
+
+    Assert(
+      triangulation != nullptr,
+      ExcMessage(
+        "parallel::CellWeights requires a parallel::TriangulationBase object."));
+
+    register_constant_weighting();
+  }
+
+
+
+  template <int dim, int spacedim>
+  void
+  CellWeights<dim, spacedim>::register_constant_weighting(
+    const unsigned int factor)
+  {
+    tria_listener.disconnect();
+
+    tria_listener = triangulation->signals.cell_weight.connect(
+      std::bind(&CellWeights<dim, spacedim>::weighting_callback,
+                std::placeholders::_1,
+                std::placeholders::_2,
+                std::cref(*dof_handler),
+                std::cref(*triangulation),
+                constant_weighting(factor)));
+  }
+
+
+
+  template <int dim, int spacedim>
+  void
+  CellWeights<dim, spacedim>::register_ndofs_weighting(
+    const unsigned int factor)
+  {
+    tria_listener.disconnect();
+
+    tria_listener = triangulation->signals.cell_weight.connect(
+      std::bind(&CellWeights<dim, spacedim>::weighting_callback,
+                std::placeholders::_1,
+                std::placeholders::_2,
+                std::cref(*dof_handler),
+                std::cref(*triangulation),
+                ndofs_weighting<unsigned int, unsigned int>({factor, 1})));
+  }
+
+
+
+  template <int dim, int spacedim>
+  void
+  CellWeights<dim, spacedim>::register_ndofs_squared_weighting(
+    const unsigned int factor)
+  {
+    tria_listener.disconnect();
+
+    tria_listener = triangulation->signals.cell_weight.connect(
+      std::bind(&CellWeights<dim, spacedim>::weighting_callback,
+                std::placeholders::_1,
+                std::placeholders::_2,
+                std::cref(*dof_handler),
+                std::cref(*triangulation),
+                ndofs_weighting<unsigned int, unsigned int>({factor, 2})));
+  }
+
+
+
+  template <int dim, int spacedim>
+  void
+  CellWeights<dim, spacedim>::register_custom_weighting(
+    const std::function<unsigned int(
+      const FiniteElement<dim, spacedim> &,
+      const typename hp::DoFHandler<dim, spacedim>::cell_iterator &)>
+      custom_function)
+  {
+    tria_listener.disconnect();
+
+    const std::function<unsigned int(
+      const typename hp::DoFHandler<dim, spacedim>::cell_iterator &,
+      const FiniteElement<dim, spacedim> &)>
+      converted_function =
+        [&custom_function](
+          const typename hp::DoFHandler<dim, spacedim>::cell_iterator &cell,
+          const FiniteElement<dim, spacedim> &future_fe) -> unsigned int {
```

Comment:
I don't think you actually need to specify the return type explicitly. It doesn't hurt, but also doesn't help.

Reply:
I'd prefer to be explicit about this :-)

Reply:
I appreciate this -- I think we share that trait ;-)